http/sys_raft.go

```diff
@@ -27,7 +29,31 @@ func handleSysRaftJoinPost(core *vault.Core, w http.ResponseWriter, r *http.Requ
 		return
 	}
 
-	joined, err := core.JoinRaftCluster(context.TODO(), req.LeaderAddr, nil, req.Retry)
+	if req.LeaderAddr == "" {
+		respondError(w, http.StatusBadRequest, errors.New("leader api address is required"))
+		return
+	}
+
+	config := api.DefaultConfig()
```

Comment:
Could we keep the custom client code from the Join function, it is making it so there are no retries

Reply:
Fixed.